### ISSUE TITLE
Automatically generate block generation info from data packs

### ISSUE THREAD

**Issue by muzikbike** [way2muchnoise/JustEnoughResources#284]
1.16+ only. Admittedly this may be a bit early to ask given that world gen customization is still an experimental thing, but if it's possible to somehow hook into both vanilla's default settings (especially if they become part of the vanilla client.jar/data/minecraft/ directory in future) as well as the settings of any active data packs, it should be possible to collect and display the relevant generation data rather than require that this mod specify it.

By doing this, #239 can probably be fixed - just group each thing by the configured feature it relies on. All ore veins would be grouped due to using minecraft:ore, all disks due to using minecraft:disk, all lakes due to using minecraft:lake if that's ever a feature, and so on. (A lot of configured features are pretty weirdly named and aren't obvious to most casual players though, so some grouping of sufficiently related configured feature types (for example, everything that uses minecraft:ore, minecraft:emerald_ore and minecraft:scattered_ore would all be classified under some main ore generation heading) as well as more intuitive naming would be required.

This should also allow resources from mods to be calculated and displayed in much the same way, assuming the mods in question define said features through data rather than hardcoding them (compatibility with this mod would be a good incentive to use data definition).

**Comment by way2muchnoise** [comment]
Yeah pulling in the data from the new data driven approach would be a nice addition. It will however be a lot of internal changes, and I don't have a ton of time to spend at the moment.
